pkg/core/native/designate.go

```diff
@@ -260,14 +260,14 @@ func (s *Designate) GetDesignatedByRole(d dao.DAO, r noderoles.Role, index uint3
 	var ns NodeList
 	var bestIndex uint32
 	var resSi state.StorageItem
-	for k, si := range kvs {
-		if len(k) < 4 {
+	for _, kv := range kvs {
+		if len(kv.Key) < 4 {
 			continue
 		}
-		siInd := binary.BigEndian.Uint32([]byte(k))
+		siInd := binary.BigEndian.Uint32([]byte(kv.Key))
 		if (resSi == nil || siInd > bestIndex) && siInd <= index {
 			bestIndex = siInd
-			resSi = si
+			resSi = kv.Item
```

Comment:
This logic is here due to iterating over map. As we now have sorted slice, may be we can replace it with iterating backwards and `break`?